src/Native/Unix/System.Security.Cryptography.Native/pal_ssl.cpp

```diff
@@ -414,13 +414,27 @@ extern "C" int32_t CryptoNative_GetSslConnectionInfo(SSL* ssl,
     return 0;
 }
 
+// Controls if ERR_clear_error is going to be called before SSL_write/SSL_read
+static bool g_forceOpenSslErrorQueueCleanupBeforeWriteRead = false;
+
+extern "C" void CryptoNative_ForceOpenSslErrorQueueCleanupBeforeWriteRead()
+{
+    g_forceOpenSslErrorQueueCleanupBeforeWriteRead = true;
+}
+
 extern "C" int32_t CryptoNative_SslWrite(SSL* ssl, const void* buf, int32_t num)
 {
+    if (g_forceOpenSslErrorQueueCleanupBeforeWriteRead)
+        ERR_clear_error();
+
     return SSL_write(ssl, buf, num);
 }
 
 extern "C" int32_t CryptoNative_SslRead(SSL* ssl, void* buf, int32_t num)
 {
+    if (g_forceOpenSslErrorQueueCleanupBeforeWriteRead)
+        ERR_clear_error();
```

Comment:
Please add the explicit braces.  The only things in this file (and probably most of this library) which use an unbraced if are stacked `if`->`return`s